src/main/java/seedu/duke/command/EditTransactionCommand.java

```diff
@@ -0,0 +1,169 @@
+        }
+
+        if (getArg(DATE_ARG) != null && !getArg(DATE_ARG).isBlank()) {
+            throw new DukeException(DATE_EDIT);
+        }
+
+        if (getArgs().size() == 1) {
+            throw new DukeException(MISSING_EDIT);
+        }
+
+        if (getArgs().size() > 2) {
+            throw new DukeException(TOO_MANY_ARGUMENTS);
+        }
+
+        if (getArg(AMOUNT_ARG) != null && !getArg(AMOUNT_ARG).isBlank()) {
+            Double amount = Parser.parseNonNegativeDouble(getArg(AMOUNT_ARG));
+            if (amount == null) {
+                throw new DukeException(BAD_AMOUNT);
+            }
+        }
+    }
+
+    private boolean isInteger(String description) {
+        try {
+            Integer.parseInt(description);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void editTransaction(Ui ui) throws DukeException {
+        String type = getArg("type").toLowerCase();
+        int maxSize = getTransactionMaxSize(type);
+        int idx = parseIdx(maxSize) - 1; //-1 due to 0 based indexing for arraylist
+        assert idx >= 0 : "Index should be a valid integer greater than 0";
+
+        boolean isSuccess = false;
+        String transactionDescription = "";
+        if (type.equals("in")) {
+            Income income = StateManager.getStateManager().getIncome(idx);
+            transactionDescription = StateManager.getStateManager().getIncome(idx)
+                    .getTransaction().getDescription();
+            if (getArg(DESCRIPTION_ARG) != null && !getArg(DESCRIPTION_ARG).isBlank()) {
+                String originalDescription = income.getTransaction().getDescription();
+                StateManager.getStateManager().getIncome(idx)
+                        .getTransaction().setDescription(getArg(DESCRIPTION_ARG));
+                String newDescription = income.getTransaction().getDescription();
+                isSuccess = !originalDescription.equals(newDescription);
+            } else if (getArg(AMOUNT_ARG) != null && !getArg(AMOUNT_ARG).isBlank()) {
+                Double originalAmount = income.getTransaction().getAmount();
+                StateManager.getStateManager().getIncome(idx)
+                        .getTransaction().setAmount(Parser.parseNonNegativeDouble(getArg(AMOUNT_ARG)));
+                Double newAmount = income.getTransaction().getAmount();
+                isSuccess = !originalAmount.equals(newAmount);
+            } else if (getArg(GOAL_ARG) != null && !getArg(GOAL_ARG).isBlank()) {
+                String originalGoalDescription = income.getGoal().getDescription();
+                StateManager.getStateManager().getIncome(idx)
+                        .getGoal().setDescription(getArg(GOAL_ARG));
+                String newGoalDescription = income.getGoal().getDescription();
+                isSuccess = !originalGoalDescription.equals(newGoalDescription);
+            }
```

Comment:
Editing the goal of a transaction should perform a lookup for the new goal, it should not edit the description of the existing goal referenced by the Transaction.

This will result in the ability to have multiple goals with the same name, which is not an allowed behaviour.